.github/workflows/build_and_test.yaml

```
@@ -1,21 +1,34 @@
 name: Build and test
-on: [workflow_dispatch, pull_request]
+on:
+  pull_request: {}
+  workflow_dispatch:
+    inputs:
+      runner:
+        description: Which runner to use
+        type: choice
+        default: 'ubuntu-latest'
```

```suggestion
        default: 'ubuntu-22.04'
```
I would prefer well defined runner version instead of latest

---

Good point!